src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbol.cs

```diff
@@ -27,5 +30,586 @@ protected static void ReportBadRefToken(TypeSyntax returnTypeSyntax, DiagnosticB
                 diagnostics.Add(ErrorCode.ERR_UnexpectedToken, refKeyword.GetLocation(), refKeyword.ToString());
             }
         }
+
+        public override FlowAnalysisAnnotations ReturnTypeFlowAnalysisAnnotations =>
```

Comment:
>public override FlowAnalysisAnnotations ReturnTypeFlowAnalysisAnnotations => [](start = 8, length = 76)

It looks like moving all this code to this class affect LambdaSymbol in an undesired way.

Reply:
After discussing offline, will try introducing a class `SourceMethodSymbolWithAttributes` which `LocalFunctionSymbol` and `SourceMemberMethodSymbol` will inherit from.